app/server_test.go

```diff
@@ -365,93 +367,108 @@ func TestSentry(t *testing.T) {
 	client := &http.Client{Timeout: 5 * time.Second, Transport: &http.Transport{
 		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
 	}}
-	data1 := make(chan bool, 1)
+	testDir, _ := fileutils.FindDir("tests")
 
-	server1 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		t.Log("Received sentry request for some reason")
-		data1 <- true
-	}))
-	defer server1.Close()
+	t.Run("sentry is disabled, should not receive a report", func(t *testing.T) {
+		data := make(chan bool, 1)
 
-	// make sure we don't report anything when sentry is disabled
-	_, port, _ := net.SplitHostPort(server1.Listener.Addr().String())
-	SENTRY_DSN = fmt.Sprintf("http://test:test@localhost:%s/123", port)
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			t.Log("Received sentry request for some reason")
+			data <- true
+		}))
+		defer server.Close()
 
-	testDir, _ := fileutils.FindDir("tests")
-	s, err := NewServer(func(server *Server) error {
-		configStore, _ := config.NewFileStore("config.json", true)
-		store, _ := config.NewStoreFromBacking(configStore)
-		server.configStore = store
-		server.UpdateConfig(func(cfg *model.Config) {
-			*cfg.ServiceSettings.ListenAddress = ":0"
-			*cfg.LogSettings.EnableSentry = false
-			*cfg.ServiceSettings.ConnectionSecurity = "TLS"
-			*cfg.ServiceSettings.TLSKeyFile = path.Join(testDir, "tls_test_key.pem")
-			*cfg.ServiceSettings.TLSCertFile = path.Join(testDir, "tls_test_cert.pem")
+		// make sure we don't report anything when sentry is disabled
+		_, port, _ := net.SplitHostPort(server.Listener.Addr().String())
+		dsn, err := sentry.NewDsn(fmt.Sprintf("http://test:test@localhost:%s/123", port))
+		require.NoError(t, err)
+		SENTRY_DSN = dsn.String()
+
+		s, err := NewServer(func(server *Server) error {
+			configStore, _ := config.NewFileStore("config.json", true)
+			store, _ := config.NewStoreFromBacking(configStore)
+			server.configStore = store
+			server.UpdateConfig(func(cfg *model.Config) {
+				*cfg.ServiceSettings.ListenAddress = ":0"
+				*cfg.LogSettings.EnableSentry = false
+				*cfg.ServiceSettings.ConnectionSecurity = "TLS"
+				*cfg.ServiceSettings.TLSKeyFile = path.Join(testDir, "tls_test_key.pem")
+				*cfg.ServiceSettings.TLSCertFile = path.Join(testDir, "tls_test_cert.pem")
+				*cfg.LogSettings.EnableDiagnostics = true
```

Comment:
What was the actual bug here? Was it that we weren't setting `EnableDiagnostics`? But by default it's always true.

Or we weren't using `sentry.NewDsn` to form the dsn?

In both cases, I don't see how it was making the test flaky. :thinking:

Reply:
From my inspection it seemed to me that timeout may be insufficient or the server actually never started. That's why I added a check if we are getting the expected error.

Since I'm unable to reproduce this locally, I added some additional checks to my lucky guesses.